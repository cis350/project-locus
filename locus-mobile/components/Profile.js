/* eslint-disable global-require */
import React from 'react';
import {
  View, Text, StyleSheet, Image,
} from 'react-native';

export default function Profile({ user }) {
  return (
    <View style={styles.container}>
      <Text style={{ fontSize: 40 }}>Profile</Text>
      <View style={styles.profileContainer}>
        <Image source={require('../assets/default-profile.jpg')} style={styles.img} />
        <Text>{`${user.firstName} ${user.lastName}`}</Text>
        <Text>Year: {user.year}</Text>
        <Text>Major: {user.major}</Text>
        <Text>Email: {user.email}</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignItems: 'center',
  },
  profileContainer: {
    backgroundColor: '#bfe48c',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 25,
    paddingVertical: 25,
    borderRadius: 10,
    marginVertical: 10,
  },
  img: {
    width: 300,
    height: 300,
  },
});
