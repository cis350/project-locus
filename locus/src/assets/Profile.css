.profile-text {
  text-align: left;
  max-width: 100%;
  max-height: 100%;
}

.profile-container {
  /* flex-direction: row; */
  /* justify-content: flex-start; */
  padding: 100px 50px 100px 50px;
  background-color: #b5e48c;
  border-radius: 10px;
  margin-left: 0px;
}

.profile-button {
  color: white;
  margin-top: 30px;
  background-color: #6a9b72;
  max-width: 100px;
  margin-left: 0px;
}

.profile-pic {
  margin-left: 0px;
  max-width: 100%;
  max-height: 100%;
}
