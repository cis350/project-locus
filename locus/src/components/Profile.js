/* eslint-disable react/jsx-one-expression-per-line */
import React, { useState, useEffect } from 'react';
import { getUser } from '../modules/api';
import '../assets/Profile.css';

export default function Profile({ userEmail }) {
  const [user, setUser] = useState(undefined);

  useEffect(() => {
    async function initialize() {
      setUser(await getUser(userEmail));
    }
    initialize();
  }, []);

  if (!user) return <div>Loading...</div>;
  return (
    <div className="container">
      <h1 className="profile-text">Profile</h1>
      <div className="container profile-container">
        <div className="row justify-content-between">
          <div className="col-4 d-flex flex-column">
            <img className="profile-pic" src="https://t4.ftcdn.net/jpg/00/64/67/63/360_F_64676383_LdbmhiNM6Ypzb3FM4PPuFP9rHe7ri8Ju.jpg" alt="profile" />
          </div>
          <div className="col-7">
            <h3 className="profile-text">{`${user.firstName} ${user.lastName}`}</h3>
            <p className="profile-text">Year: {user.year}</p>
            <p className="profile-text">Major: {user.major}</p>
            <p className="profile-text">Email: {user.email}</p>
          </div>
        </div>
      </div>
    </div>
  );
}
