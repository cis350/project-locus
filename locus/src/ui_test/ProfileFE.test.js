/**
* @jest-environment jsdom
*/
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Profile from '../components/Profile';

const user = {
  id: 1, name: 'Dustin Fang', year: 2024, major: 'Computer Science', email: 'fdustin@seas.upenn.edu',
};

test('member profile test', () => {
  render(<Profile id={user.id} />);
  const linkElement = screen.getByText('Member Profile');
  expect(linkElement).toBeInTheDocument();
});

test('name test', () => {
  render(<Profile id={user.id} />);
  const linkElement = screen.getByText('Dustin Fang');
  expect(linkElement).toBeInTheDocument();
});

test('year test', () => {
  render(<Profile id={user.id} />);
  const linkElement = screen.getByText('Year: 2024');
  expect(linkElement).toBeInTheDocument();
});

test('major test', () => {
  render(<Profile id={user.id} />);
  const linkElement = screen.getByText('Major: Computer Science');
  expect(linkElement).toBeInTheDocument();
});

test('email test', () => {
  render(<Profile id={user.id} />);
  const linkElement = screen.getByText('Email: fdustin@seas.upenn.edu');
  expect(linkElement).toBeInTheDocument();
});

test('message test', () => {
  render(<Profile id={user.id} />);
  const linkElement = screen.getByText('Message');
  expect(linkElement).toBeInTheDocument();
});

test('image test', () => {
  render(<Profile id={user.id} />);
  const linkElement = screen.getByText('Message');
  expect(linkElement).toBeInTheDocument();
});
