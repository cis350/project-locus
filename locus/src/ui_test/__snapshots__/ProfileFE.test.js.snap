// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<div
  className="container"
>
  <h1
    className="profile-text"
  >
    Profile
  </h1>
  <div
    className="container profile-container"
  >
    <div
      className="row justify-content-between"
    >
      <div
        className="col-4 d-flex flex-column"
      >
        <img
          alt="profile"
          className="profile-pic"
          src="https://t4.ftcdn.net/jpg/00/64/67/63/360_F_64676383_LdbmhiNM6Ypzb3FM4PPuFP9rHe7ri8Ju.jpg"
        />
        <button
          className="btn profile-button"
          onClick={[Function]}
          type="button"
        >
          <strong>
            Message
          </strong>
        </button>
      </div>
      <div
        className="col-7"
      >
        <h3
          className="profile-text"
        >
          Dustin Fang
        </h3>
        <p
          className="profile-text"
        >
          Year: 
          2024
        </p>
        <p
          className="profile-text"
        >
          Major: 
          Computer Science
        </p>
        <p
          className="profile-text"
        >
          Email: 
          fdustin@seas.upenn.edu
        </p>
      </div>
    </div>
  </div>
</div>
`;
